src/quacc/utils/files.py

```diff
@@ -69,9 +69,57 @@ def copy_decompress_files(
         if f_path.is_file():
             copy(f_path, Path(destination, f_path.name))
             decompress_file(Path(destination, f_path.name))
+        elif f_path.is_dir():
+            copy_decompress_files_from_dir(f_path, destination)
         else:
             warnings.warn(f"Cannot find file {f_path}", UserWarning)
 
+def copy_decompress_tree(
+    source_files: dict[str, str | Path | list[str | Path]], destination: str | Path
+) -> None:
+    """
+    Copy and decompress files from source to destination.
+    This function respects the directory tree.
```

Comment:
It would be very helpful if you could add unit tests for this function, independent of Espresso. I will probably want to refactor this at some point but would be hesitant to do so without the peace of mind of simple-to-run tests.